src/main/java/com/lishid/openinv/ConfigUpdater.java

```diff
@@ -0,0 +1,161 @@
+package com.lishid.openinv;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.bukkit.Material;
+import org.bukkit.configuration.ConfigurationSection;
+
+import com.lishid.openinv.utils.UUIDUtil;
+
+public class ConfigUpdater {
+    private final OpenInv plugin;
+
+    private static final int CONFIG_VERSION = 2;
+
+    public ConfigUpdater(OpenInv plugin) {
+        this.plugin = plugin;
+    }
+
+    private int getConfigVersion() {
+        return plugin.getConfig().getInt("config-version", 1);
+    }
+
+    private boolean isConfigOutdated() {
+        return getConfigVersion() < CONFIG_VERSION;
+    }
+
+    public void checkForUpdates() {
+        if (isConfigOutdated()) {
+            plugin.getLogger().info("[Config] Update found! Performing update...");
+            updateConfig();
```

Comment:
I'd suggest comparing against specific versions and running specific upgrade commands for each version.
Basically upgradeConfig1to2(), and in the future, upgradeConfig2to3(), etc. This ensures that you're not risking running upgrade 1 to 2 more than once and error out because the migration already happened.

Something like if(version < 2) { upgrade1to2(); }
Make sure you set version to 2 and not CONFIG_VERSION in the process.